### PR TITLE
A demostration of the new way of calculating account ids

### PR DESCRIPTION
**Pull request by dong77**
https://github.com/Loopring/protocols/issues/1218

### CODE REVIEW

File: packages/loopring_v3/contracts/impl/libexchange/ExchangeAccounts.sol [hunk 1]
```diff
@@ -140,6 +140,18 @@ library ExchangeAccounts
         );
     }
 
+    function calculateAccountId(uint24 accountIdx)
+        public
+        pure
+        returns (uint24 accountID)
+    {
+        uint24 idx = accountIdx;
+        for (uint i = 0; i < 192; i++) {
```
Comment (Brechtpd) [inline packages/loopring_v3/contracts/impl/libexchange/ExchangeAccounts.sol:149]: 192?

**Comment by Brechtpd** [comment]
Another possible way to do this is to keep the nicely incrementing accountIDs, but just read the bits of the value flipped when using it as an address. So instead of starting at the LSB, we start at the MSB. Though perhaps that would be more confusing.

File: packages/loopring_v3/contracts/impl/libexchange/ExchangeAccounts.sol (hunk 1 above)
Comment (dong77) [reply packages/loopring_v3/contracts/impl/libexchange/ExchangeAccounts.sol]: my bad.

File: packages/loopring_v3/contracts/impl/libexchange/ExchangeAccounts.sol [hunk 2]
```diff
@@ -140,6 +140,18 @@ library ExchangeAccounts
         );
     }
 
+    function calculateAccountId(uint24 accountIdx)
```
Comment (dong77) [inline packages/loopring_v3/contracts/impl/libexchange/ExchangeAccounts.sol:143]: @Brecht, do you think it's a good idea to introduce a concept of TreeSharding

interface TreeSharding
{
     function accountIDFromAccountIndex(uint24 accountIndex) returns (uint24 accountID);
}

**Pull request closed without merging**